langlib/lang.value/src/main/ballerina/value.bal

```diff
@@ -50,7 +50,7 @@ public isolated function clone(CloneableType v) returns CloneableType = @java:Me
 #
 # + v - source value
 # + return - immutable clone of `v`
-public isolated function cloneReadOnly(AnydataType v) returns AnydataType = @java:Method {
+public isolated function cloneReadOnly(CloneableType  v) returns CloneableType = @java:Method {
```

Comment:
Shouldn't the `testStructureWithErrorValueFreeze` test in https://github.com/ballerina-platform/ballerina-lang/blob/master/tests/jballerina-unit-test/src/test/resources/test-src/expressions/builtinoperations/freeze-and-isfrozen.bal#L458 have failed previously, now that `error` is not part of `anydata`?

Can we add a test for the same? Attempting to assign a structure with errors to anydata?

The following does not produce an error on slp9-SNAPSHOT.

```ballerina
public function main() {
    (anydata|error)[] x = [error("")];
    anydata y = x;
}
```

Reply:
Created an issue for this 
https://github.com/ballerina-platform/ballerina-lang/issues/28040